src/components/timeline/TimelineCursors.svelte

```diff
@@ -9,30 +11,98 @@
   export let mouseOver: MouseOver;
   export let histogramCursorTime: Date | null = null;
   export let xScaleView: ScaleTime<number, number> | null = null;
+  export let verticalGuides: VerticalGuide[] = [];
+  export let timelineId: number;
+
+  type ComputedVerticalGuide = { id: number; label: Label; maxWidth: number; x: number };
 
   $: onCursorEnableChange(cursorEnabled);
   $: onMouseOver(mouseOver);
   $: onHistogramCursorTime(histogramCursorTime);
-  $: if (xScaleView) {
-    hideCursor();
-  }
+  $: onVerticalGuidesChange(verticalGuides, xScaleView, drawWidth);
 
-  let cursorDiv: HTMLElement = null;
-  let cursorLabelDiv: HTMLElement = null;
   let offsetX: number = -1;
-  let raf: number = null;
+  let cursorX: number = 0;
+  let cursorMaxWidth: number = 0;
+  let cursorDOY: string = '';
+  let computedVerticalGuides: ComputedVerticalGuide[] = [];
+  let cursorWithinView = true;
+
+  /**
+   * Sort vertical guides in time order descending.
+   */
+  function sortVerticalGuides(verticalGuides: VerticalGuide[]): VerticalGuide[] {
+    return [...verticalGuides].sort((a: VerticalGuide, b: VerticalGuide) => {
+      const aTime = getUnixEpochTime(a.timestamp);
+      const bTime = getUnixEpochTime(b.timestamp);
+      return bTime - aTime;
+    });
+  }
+
+  function onVerticalGuidesChange(verticalGuides: VerticalGuide[], xScaleView, drawWidth) {
```

Comment:
I'm getting a svelte check warning here to add types to the params:

`xScaleView: ScaleTime<number, number> | null`
`drawWidth: number`

Reply:
Oh thanks for pointing that out, I wasn't aware of the `svelte-check` output previously! Will patch these up and make sure to keep an eye on it going forward.